spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -195,8 +196,10 @@ protected Statistics estimateStatistics(Snapshot snapshot) {
     if (readConf.reportColumnStats() && cboEnabled) {
       colStatsMap = Maps.newHashMap();
       List<StatisticsFile> files = table.statisticsFiles();
-      if (!files.isEmpty()) {
-        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
+      Optional<StatisticsFile> file =
+          files.stream().filter(f -> f.snapshotId() == snapshot.snapshotId()).findFirst();
```

Comment:
Could we have multiple files with the same snapshotId? Or `findAny` is enough?

Reply:
As per java implementation, it is always a single stats file per snapshot id. 

https://github.com/apache/iceberg/blob/695374d8bfbd22f4347cfe43b54a93c41b88f166/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1369

I followed similar design while working on partition stats too.

Reply:
So we could have a theoretical performance boost from `findAny`

Reply:
The spec doesn't actually say that there should only be *one* statistics file per snapshot. This happens to be how it is implemented in Java. The spec simply allows for multiple statistics files.
I was thinking about the problem of tracking orphaned statistics files when they are recomputed. One idea I had was to keep replaced statistics files (for a snapshot) still in the list (as long as the files are tracked in metadata we can clean up unused ones), but to keep the newest one before others. Hence `findFirst`. It was just an idea (and honestly not one I'm seriously considering).
In any case, I do not think that `findAny` is faster than `findFirst` here.